lib/mail.ex

```diff
@@ -119,6 +119,9 @@ defmodule Mail do
   def put_attachment(%Mail.Message{multipart: true} = message, path) when is_binary(path),
     do: Mail.Message.put_part(message, Mail.Message.build_attachment(path))
 
+  def put_attachment(%Mail.Message{multipart: true} = message, path) when is_tuple(path),
```

Comment:
I don't think `path` is the current variable name here. I think it would be best to pattern match the tuple rather than use a guard. This way we can ensure the correct tuple size is used:

```elixir
def put_attachment(%Mail.Message{multipart: true} = message, {filename, data})
  do: Mail.Message.put_part(message, Mail.Message.build_attachment({filename, data}))
```